From 5c55fae8165a746006b1ae8bdbf919fbf6bfdb6f Mon Sep 17 00:00:00 2001
From: Thierry Vignaud <thierry.vignaud@gmail.com>
Date: Mon, 17 Dec 2012 13:58:44 +0100
Subject: [PATCH] display the actual unknown option instead of '?'

'?' is returned by getopt when option is unknown, making hard to
pinpoint the actual bogus option...
---
 rpmio/macro.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rpmio/macro.c b/rpmio/macro.c
index f7954b3..5362512 100644
--- a/rpmio/macro.c
+++ b/rpmio/macro.c
@@ -823,7 +823,7 @@ grabArgs(MacroBuf mb, const rpmMacroEntry me, const char * se,
 	char *name = NULL, *body = NULL;
 	if (c == '?' || strchr(opts, c) == NULL) {
 	    rpmlog(RPMLOG_ERR, _("Unknown option %c in %s(%s)\n"),
-			(char)c, me->name, opts);
+			(char)optopt, me->name, opts);
 	    goto exit;
 	}
 
-- 
1.8.0.2

